lib/connect.c

```diff
@@ -629,7 +631,7 @@ bool Curl_addr2string(struct sockaddr *sa, curl_socklen_t salen,
 #ifdef ENABLE_IPV6
   struct sockaddr_in6 *si6 = NULL;
 #endif
-#if defined(HAVE_SYS_UN_H) && defined(AF_UNIX)
+#if defined(HAVE_SOCKADDR_UN) && defined(AF_UNIX)
```

Comment:
`HAVE_SOCKADDR_UN` is only set in cmake and Windows builds, excluding all the configure builds!